app/src/main/java/com/sopt/now/ui/signup/SignUpActivity.kt

```diff
@@ -1,83 +1,75 @@
-        intent.putExtra(
-            TAG_USER, User(
-                id = binding.etSignupId.text.toString(),
-                password = binding.etSignupPassword.text.toString(),
-                nickname = binding.etSignupNickname.text.toString(),
-                mbti = binding.etSignupMBTI.text.toString()
-            )
-        )
-        setResult(RESULT_OK, intent)
-        finish()
-
-    }
-
-    private fun validateSignUp() =
-        validateID() && validatePassword() && validateNickName() && validateMBTI()
+    private fun initObserver() {
+        viewModel.liveData.observe(this) { state ->
+            when (state) {
+                is UiState.Loading -> {
 
-    private fun validateID(): Boolean {
-        require(binding.etSignupId.text.length in 6..10) {
-            toast(VALIDATE_ID)
-            return false
+                }
+                is UiState.Success -> {
+                    toast("회원가입 성공 userid = ${state.data.userid} 입니다!")
+                    navToLogin(state.data)
+                }
+                is UiState.Error -> {
+                    toast(state.errorMessage)
+                }
+            }
         }
-        return true
     }
 
-    private fun validatePassword(): Boolean {
-        require(binding.etSignupPassword.text.length in 8..12) {
-            toast(VALIDATE_PASSWORD)
-            return false
+    private fun initButton() {
+        binding.btnLogin.setOnClickListener {
+            signUpEvent()
         }
-        return true
     }
 
-    private fun validateNickName(): Boolean {
-        require(binding.etSignupNickname.text.isNotBlank()) {
-            toast(VALIDATE_NICKNAME)
-            return false
-        }
-        return true
+    private fun signUpEvent() {
+        viewModel.signUp(getSignUpRequestDto())
     }
 
-    private fun validateMBTI(): Boolean {
-        require(binding.etSignupMBTI.text.length == 4) {
-            toast(VALIDATE_MBTI)
-            return false
-        }
-        return true
+    private fun navToLogin(user: User) {
+        val intent = Intent(this@SignUpActivity, LoginActivity::class.java)
+        intent.putExtra(TAG_USER, user)
+        setResult(RESULT_OK, intent)
+        finish()
     }
 
-    companion object {
-        const val VALIDATE_ID = "ID입력 조건에 맞지 않습니다."
-        const val VALIDATE_PASSWORD = "비밀번호 입력 조건에 맞지 않습니다."
-        const val VALIDATE_NICKNAME = "닉네임 입력 조건에 맞지 않습니다."
-        const val VALIDATE_MBTI = "MBTI 입력 조건에 맞지 않습니다."
+    private fun getSignUpRequestDto(): RequestSignUpDto {
+        val id = binding.etSignupId.text.toString()
+        val password = binding.etSignupPassword.text.toString()
+        val nickname = binding.etSignupNickname.text.toString()
+        val mbti = binding.etSignupNumbers.text.toString()
+        return RequestSignUpDto(
+            authenticationId = id,
+            password = password,
+            nickname = nickname,
+            phone = mbti
+        )
```

Comment:
```suggestion
        return RequestSignUpDto(
            authenticationId = binding.etSignupId.text.toString(),
            password = binding.etSignupPassword.text.toString(),
            nickname = binding.etSignupNickname.text.toString(),
            phone = binding.etSignupNumbers.text.toString()
        )
```
이렇게 불필요한 변수 선언 없이 사용할 수 있겠다는 생각도 듭니다!

Reply:
오.. 덕분에 아래와 같이 수정했습니다 감사합니다
![image](https://github.com/NOW-SOPT-ANDROID/myeongseok_kim/assets/75840431/6f1cf1a0-bbaa-4ab2-b5a7-c6fb13992828)